app/middleware/getGp.js

```diff
@@ -0,0 +1,31 @@
+const MongoClient = require('mongodb').MongoClient;
+const co = require('co');
+const log = require('../lib/logger');
+const config = require('../../config/config').mongodb;
+
+function getGp(req, res, next) {
+  const connectionString = config.connectionString;
+
+  co(function* () {
```

Comment:
Rather than using the co library, the mongoclient library has built in support for promises, so you can replace the 
 const db = yield MongoClient.connect(connectionString);
with 
MongoClient.connect(connectionString).then((db => {}).catch((err) => {...})

Anywhere there is a yield from a mongo call a promise can be used.
Airbnb's current JS style guides recommend not to use generators, but I believe that's more for transpiling to es5 for client side code.

I think this perhaps a larger conversation about whether to use generators or promises for our asynchronous approach.